packages/reports/src/reports/blocks/BlockPreview.jsx

```diff
@@ -36,40 +38,56 @@ function BlockPreview(props) {
    * Use the "blockStateContent" prop to provide override values if you want to show a
    * live preview of unsaved changes.
    */
-  const blockContent = blockStateContent || getBlockContent(block, locale);
+  const blockContent =  BLOCK_CONFIG[block.type].renderPreviewOnEdit && blockStateContent || getBlockContent(block, locale);
 
   /* redux */
   const formatterFunctions = useFormatters(locale);
 
-  const {content, error, log} = useMemo(() => {
-    let payload = {};
-    // if a Block-specific preview adapter function exists, use that to build payload
-    if (PreviewAdapters[block.type] && typeof PreviewAdapters[block.type] === "function") {
+  const [content, error, log] = useMemo(() => {
```

Comment:
For my own edification, is there a performance or common practice reason for choosing an array instead of an object here?

Reply:
Not a performance reason really -- it just allows you to destructure the variables under whatever name you want them rather than having to remap named variables in an object.